### PR TITLE
Test to verify diag_encl - v3

### PR DESCRIPTION
**Pull request by Pavithra1602**
This test verifies '-d' option of diag_encl. It checks whether single xml file is generated when "diag_encl" command is run multiple times and validates xml content.

Signed-off-by: Pavithra <pavrampu@linux.vnet.ibm.com>

### CODE REVIEW

File: generic/diag_encl.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import os
+import xml.etree.ElementTree
+from avocado import Test
+from avocado import main
+from avocado.utils import process, distro
+from avocado.utils import genio
+from avocado.utils.software_manager import SoftwareManager
+
+
+class diag_encl(Test):
```
Comment (harish-24) [inline generic/diag_encl.py]: Please adhere to python standards. Could be `DiagEncl`.

File: generic/diag_encl.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    This test checks various options of diag_encl tool:
+    """
+    is_fail = 0
+
+    def run_cmd(self, cmd):
+        if (process.run(cmd, ignore_status=True, sudo=True, shell=True)).exit_status:
+            self.is_fail += 1
+        return
+
+    def run_cmd_out(self, cmd):
```
Comment (harish-24) [inline generic/diag_encl.py]: Can you add `@staticmethod` before this method? As none of the Test class objects and methods are used here.

File: generic/diag_encl.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        return process.system_output(cmd, shell=True, ignore_status=True, sudo=True)
+
+    def setUp(self):
+        if "ppc" not in distro.detect().arch:
+            self.cancel("supported only on Power platform")
+        sm = SoftwareManager()
```
Comment (harish-24) [inline generic/diag_encl.py]: Minor suggestion: use at-least three characters variable name.

File: generic/diag_encl.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            self.run_cmd("diag_encl -d")
+        xml_file_path = os.path.join(diag_path, '*diskAnalytics*')
+        no_of_files = self.run_cmd_out("ls -lrt %s | wc -l" % xml_file_path).strip()
+        if no_of_files == '0':
+            self.fail("xml file not generated")
+        if no_of_files > '1':
+            self.fail("multiple xml files are generated")
+        xml_file = self.run_cmd_out("ls %s" % diag_path).strip()
+        xml_file = os.path.join(diag_path, xml_file)
+        e = xml.etree.ElementTree.parse(xml_file).getroot()
```
Comment (harish-24) [inline generic/diag_encl.py]: same here

**Comment by Pavithra1602** [comment]
Raised New PR.

**Pull request closed without merging**